src/components/side_panel/pivot/pivot_measure_display_panel/pivot_measure_display_panel_store.ts

```diff
@@ -0,0 +1,184 @@
+    super(get);
+    this.measureDisplay = initialMeasure.display || { type: "no_calculations" };
+  }
+
+  updateMeasureDisplayType(measureDisplayType: PivotMeasureDisplayType) {
+    this.updatePivotMeasureDisplay(
+      this.getMeasureDisplay(
+        measureDisplayType,
+        this.measureDisplay.fieldNameWithGranularity,
+        this.measureDisplay.value
+      )
+    );
+  }
+
+  updateMeasureDisplayField(fieldNameWithGranularity: string) {
+    this.updatePivotMeasureDisplay(
+      this.getMeasureDisplay(
+        this.measureDisplay.type,
+        fieldNameWithGranularity,
+        this.measureDisplay.value
+      )
+    );
+  }
+
+  updateMeasureDisplayValue(value: string) {
+    this.updatePivotMeasureDisplay(
+      this.getMeasureDisplay(
+        this.measureDisplay.type,
+        this.measureDisplay.fieldNameWithGranularity,
+        value
+      )
+    );
+  }
+
+  private updatePivotMeasureDisplay(newDisplay: PivotMeasureDisplay) {
+    const pivotDefinition = deepCopy(this.model.getters.getPivotCoreDefinition(this.pivotId));
+    const measureIndex = this.getMeasureIndex(this.initialMeasure.id, pivotDefinition);
+    const newMeasure = { ...pivotDefinition.measures[measureIndex], display: newDisplay };
+    pivotDefinition.measures[measureIndex] = newMeasure;
+    const result = this.model.dispatch("UPDATE_PIVOT", {
+      pivot: pivotDefinition,
+      pivotId: this.pivotId,
+    });
+    if (result.isSuccessful) {
+      this.measureDisplay = newDisplay;
+    }
+  }
+
+  private getMeasureDisplay(
+    measureDisplayType: PivotMeasureDisplayType,
+    fieldNameWithGranularity: string | undefined,
+    value: string | boolean | number | undefined
+  ): PivotMeasureDisplay {
+    switch (measureDisplayType) {
+      case "no_calculations":
+      case "%_of_grand_total":
+      case "%_of_col_total":
+      case "%_of_row_total":
+      case "%_of_parent_row_total":
+      case "%_of_parent_col_total":
+      case "index":
+        return { type: measureDisplayType };
+      case "%_of_parent_total":
+      case "running_total":
+      case "%_running_total":
+      case "rank_asc":
+      case "rank_desc":
+        if (!fieldNameWithGranularity) {
+          fieldNameWithGranularity = this.fields[0]?.nameWithGranularity;
+        }
+        return { type: measureDisplayType, fieldNameWithGranularity };
+      case "%_of":
+      case "difference_from":
+      case "%_difference_from":
+        if (!fieldNameWithGranularity) {
+          fieldNameWithGranularity = this.fields[0]?.nameWithGranularity;
+        }
+        const possibleValues = this.getPossibleValues(fieldNameWithGranularity);
+        if (value === undefined || !possibleValues.find((v) => v.value === value)) {
+          value = "(previous)";
+        }
+        return {
+          type: measureDisplayType,
+          fieldNameWithGranularity,
+          value: value ?? "(previous)",
```

Comment:
Maybe use a constant `PREVIOUS_VALUE` and `NEXT_VALUE` anywhere in the code. With "(previous)" and "(next)", I have to ask myself everywhere if the value should be translated or not.